src/rpc/blockchain.cpp

```diff
@@ -1138,7 +1144,25 @@ UniValue getblockchaininfo(const UniValue &params, bool fHelp)
         "verificationprogress", Checkpoints::GuessVerificationProgress(Params().Checkpoints(), chainActive.Tip()));
     obj.pushKV("initialblockdownload", IsInitialBlockDownload());
     obj.pushKV("chainwork", chainActive.Tip()->nChainWork.GetHex());
+    obj.pushKV("size_on_disk", CalculateCurrentUsage());
     obj.pushKV("pruned", fPruneMode);
+    if (fPruneMode)
+    {
+        CBlockIndex *block = chainActive.Tip();
+        assert(block);
+        {
+            READLOCK(cs_mapBlockIndex);
+            while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
+            {
+                block = block->pprev;
+            }
+        }
+
+        if (block != nullptr)
+            obj.pushKV("pruneheight", block->nHeight);
```

Comment:
else
  obj.pushKV("pruneheight", 0);

We should return this value consistently if pruning is on so that programs that use this val don't have to write conditional code.

Reply:
agreed, will fix